pandas/core/frame.py

```diff
@@ -5742,14 +5749,18 @@ class    max    type
         if not drop:
             to_insert: Iterable[tuple[Any, Any | None]]
             if isinstance(self.index, MultiIndex):
-                names = [
-                    (n if n is not None else f"level_{i}")
-                    for i, n in enumerate(self.index.names)
-                ]
+                if not names:
```

Comment:
all of this logic should be in an Index method itself. We prob don't have anything exactly like this, but could add a function to Index.get_default_index_names(names=) to give you this

Reply:
I think this can be done easily for a new method `MultiIndex.get_default_index_names`, but when selecting the new index name for `Index`, also self (that is, DataFrame) is used.

https://github.com/pandas-dev/pandas/blob/8b41541f8b17d62786fec3aa0fce1f073fd402f1/pandas/core/frame.py#L5759

Reply:
ok let's do that

Reply:
see my comment below. pls create the method and add some dedicated tests for it on Index/MultiIndex.

Reply:
So you are saying it would be fine to have two different signatures:
```
Index.get_default_index_names(self,  dataframe)
MultiIndex.get_default_index_names(self)
```
?

It would be nice if these methods had the same arguments, so one can call it without knowing whether it's an Index or a MultiIndex, but I don't see an easy way to get there.

Reply:
@jreback could you please confirm the two different signatures is the way to go?